arch/riscv/core/asm_macros.inc

```diff
@@ -0,0 +1,78 @@
+/*
+ * Assembly macros and helpers
+ *
+ * Copyright (c) 2022 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_64BIT
+	/* register-wide load/store based on ld/sd (XLEN = 64) */
+
+	.macro lr, rd, mem
```

Comment:
Doesn't `lr` collide with the actual instruction named `lr` (load reserved, from the A extension)?
Technically it might be fine since the actual instruction mnemonics are `lr.w` and `lr.d` (for 32 and 64 bit values, respectively) but it can still lead to confusion...

Reply:
Hmmm... this could be confusing indeed. I wish risc-v had an official pseudo-op for that.
What about 'lx' and 'sx' instead since this is 'xlen' based? And  nqualified registers are identified with the letter x. Or any other suggestions?

Reply:
I think that sounds like a good idea. AFAIK neither `lx` nor `sx` is used for instruction names in any current or upcoming extensions.